### ISSUE TITLE
[Feature] Allow connections to be added / removed on the fly in Connection\Group

### ISSUE THREAD

**Issue by Savageman** [hoaproject/Socket#39]
Currently if we merge connections after the run, they don't seem to be listened.

**Comment by Hywan** [comment]
Please, explain how do you do that :-). I reckon this is not possible but I woud like to know how do you run and add at the same time.

**Comment by Savageman** [comment]
Currently it's not possible with Hoa, it's a feature request: I'd like to be able to do it :)

**Comment by Hywan** [comment]
Removing is very hard because connections are “merged”, but this is still possible to undo this https://github.com/hoaproject/Socket/blob/5b136b1ba4f2309c4401de442eca38010b9c9805/Connection/Handler.php#L205-L229 and this https://github.com/hoaproject/Socket/blob/master/Server.php#L364-L392 (this one is difficult) and https://github.com/hoaproject/Socket/blob/master/Client.php#L246-L272.

Adding should already work I think.

Why would you like to remove a connection? If you close it, it's OK, isn't it? I understand your needs, I am just looking for a workaround.

**Comment by Savageman** [comment]
Maybe the Connection\Group is designed weirdly. It looks like Connection classes (both Client and Server) have code to manage multiple connections with `consider()`. Maybe this is too much responsibility and it would be easier to isolate this part into the `Connection\Handler` class only?

**Comment by Hywan** [comment]
Nop. Actually, the design seems weird because of the `select` API. We have to merge connections into a single one, this is why a connection must consider/be aware of another one. Adding a connection at runtime is easy I think, but removing one would require several tests and could cause many data races (a lot of `null` value). This is a long time development. I will modify my planning to find a period to work on this feature. Not sure I could provide it in the next months. Is it urgent?

**Comment by Savageman** [comment]
No, don't bother it's not urgent. We already developed our system with another library, but we were evaluating other libraries to migrate (the one we use it's not working with PHP 7).
